lab5/solid_principle.py

```diff
@@ -0,0 +1,142 @@
+from math import pi
+
+# --------------------------------------------------------------------------------------------
+# SRP - Single Responsibility Principle
+class Customer:
+    """
+    Create a customer.
+    """
+    def __init__(self, name, mobile):
+        self.name = name
+        self.mobile = mobile
+
+class GetCustomer(Customer):
+    """
+    Get details about the customer.
+    """
+    def get_name(self):
+        return self.name
+
+    def get_mobile(self):
+        return self.mobile
+
+def customers():
+    """
+    the function which create a customer &
+    output his name and mobiles
+    """
+    c1 = Customer("Katya", "0678543289")
+    db1 = GetCustomer(c1.name, c1.mobile)
+    print(db1.get_name() + " " + db1.get_mobile())
+
+# --------------------------------------------------------------------------------------------
+# OCP - Open-Closed Principle
+class Discount:
+    """
+    create a discount for customer
+    """
+    def __init__(self, customer, price):
+        self.customer = customer
+        self.price = price
+
+    def get_customer(self):
+        return self.customer
+
+    def get_discount(self):
+        return self.price * 0.2
+
+class VIPcustomer(Discount):
+    """
+    to extend the method get_discount() from class Discount
+    for VIP-customer to increase the discount by 2 times
+    """
+    def get_discount(self):
+        return super().get_discount() * 2
+
+def discounts():
+    """
+    the function which create a vip-customer &
+    output this customer and his discount
+    """
+    d1 = VIPcustomer("customer2", 1000)
+    print(d1.get_customer() + " - " + str(d1.get_discount()))
+
+# --------------------------------------------------------------------------------------------
+# LSP - Liskov Substitution Principle
+class Plants:
+    """
+    create a plant
+    """
+    def __init__(self, name):
+        self.name = name
+
+    def get_name(self):
+        return self.name
+
+class Flowerpot(Plants):
```

Comment:
what is the relation of a flowerpot and the plant? It's definitely not inheritance